AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -124,9 +124,15 @@ - (void)setURL:(NSURL *)URL resetToDefault:(BOOL)reset
 
   if (reset || _URL == nil) {
     self.image = _defaultImage;
-    ASPerformBlockOnMainThread(^{
-      self.currentImageQuality = 1.0;
-    });
+    if (_URL == nil) {
```

Comment:
Let's refactor this to only have one dispatch.

Above the outer if, you can do something like BOOL hasURL = (_URL != nil);.  Then you can eliminate the inner if ' else, and do ... = (hasURL ? 0.0 : 1.0);